src/pspm_convert_visangle2sps.m

```diff
@@ -91,49 +102,23 @@
             % get channel specific data
             lon = data{gx}.data;
             lat = data{gy}.data;
-%             lat = data{gy}.header.range(2)-lat;
-            
-            % first interpolate longitude to evict NaN-values
-            [bsts,outdata]=pspm_interpolate(lon);
-            if bsts ~= 1
-                warning('ID:invalid_input', 'Could not load interpolate longitude data correctly.');
-                return;
-            end
-            lon = outdata;
             
-            % first interpolate latitude to evict NaN-values
-            [bsts,outdata]=pspm_interpolate(lat);
-            if bsts ~= 1
-                warning('ID:invalid_input', 'Could not load interpolate latitude data correctly.');
-                return;
-            end
-            lat = outdata;
-            
-            %compare if length are the same
-            N = numel(lon);
-            if N~=numel(lat)
-                warning('ID:invalid_input', 'length of data in gaze_x and gaze_y is not the same');
-                return;
-            end;
-            
-            %convert lon and lat into radians
-            lon = deg2rad(lon);
-            lat = deg2rad(lat);
-            % compute distances
-            arclen = zeros(length(lat),1);
-            
-            for k = 2:length(lat)
-                lon_diff = abs(lon(k-1)-lon(k));
-                arclen(k) = atan(sqrt(((cos(lat(k))*sin(lon_diff))^2)+(((cos(lat(k-1))*sin(lat(k)))-(sin(lat(k-1))*cos(lat(k))*cos(lon_diff)))^2))/((sin(lat(k-1))*sin(lat(k)))+(cos(lat(k-1))*cos(lat(k))*cos(lon_diff))));
+            try
+                arclen = pspm_convert_visangle2sps_core(lat, lon);
+            catch
+                warning('ID:invalid_input', 'Could not calculate sps from gaze data');
+                return
             end
+
+
             % create new channel with data holding distances
-            dist_channel.data = rad2deg(arclen);
-            dist_channel.header.chantype = 'sps';
+            dist_channel.data = rad2deg(arclen) .* data{gx}.header.sr;
+            dist_channel.header.chantype = strcat('sps_', eye);
             dist_channel.header.sr = data{gx}.header.sr;
             dist_channel.header.units = 'degree';
-            
-            
-            [lsts, outinfo] = pspm_write_channel(fn, dist_channel, 'add');
+
+
+            [lsts, outinfo] = pspm_write_channel(fn, dist_channel, options.channel_action);

```

Comment:
Issue: `outinfo` is normally a structure, documentation of `pspm_write_channel` :
```
%   Outputs will be written into the info struct. The structure depends on
%   the passed action and options.
%       
%           .channel    contains channel id of added / replaced / deleted
%                       channels.
```